apps/worker/src/monitor/tx.status.monitor.service.ts

```diff
@@ -85,42 +79,47 @@ export class TxStatusMonitoringService extends WorkerHost implements OnApplicati
     // do some stuff
   }
 
-  private async crawlBlockList(txHash: Hash, epoch: string, blockList: bigint[]): Promise<BlockHash | undefined> {
-    const txReceiptPromises: Promise<BlockHash | undefined>[] = blockList.map(async (blockNumber) => {
-      const blockHash = await this.blockchainService.getBlockHash(blockNumber);
-      const block = await this.blockchainService.getBlock(blockHash);
-      const txInfo = block.block.extrinsics.find((extrinsic) => extrinsic.hash.toString() === txHash.toString());
-      this.logger.debug(`Extrinsics: ${block.block.extrinsics[0]}`);
+  private async handleMessagesFailure(jobId: string, moduleError: RegistryError): Promise<void> {
+    this.logger.debug(`Handling extrinsic failure for job ${jobId} and error ${JSON.stringify(moduleError)}`);
+    const txJob = (await this.txReceiptQueue.getJob(jobId)) as Job<ITxMonitorJob, any, string>;
+    if (!txJob) {
+      this.logger.error(`Job ${jobId} not found in tx receipt queue`);
+    }
+    const blockDelay = 1 * SECONDS_PER_BLOCK * MILLISECONDS_PER_SECOND;
 
-      if (txInfo !== undefined) {
-        this.logger.verbose(`Found tx ${txHash} in block ${blockNumber}`);
-        const at = await this.blockchainService.api.at(blockHash.toHex());
-        const events = await at.query.system.events();
-        events.subscribe((records) => {
-          records.forEach(async (record) => {
-            const { event } = record;
-            const eventName = event.section;
-            const { method } = event;
-            const { data } = event;
-            this.logger.debug(`Received event: ${eventName} ${method} ${data}`);
-            if (eventName.search('capacity') !== -1 && method.search('Withdrawn') !== -1) {
-              const capacityWithDrawn = BigInt(data[1].toString());
-              this.logger.debug(`Capacity withdrawn: ${capacityWithDrawn}`);
-              this.setEpochCapacity(epoch, capacityWithDrawn);
-            }
-          });
-        });
-        return blockHash;
+    try {
+      switch (moduleError.name) {
+        case 'TooManyMessagesInBlock':
+          // Re-try the job in the publish queue with a block delay
+          await this.publishQueue.removeRepeatableByKey(txJob.data.referencePublishJob.id);
+          await this.publishQueue.add(txJob.data.referencePublishJob.id, txJob.data.referencePublishJob, { delay: blockDelay });
+          break;
+        case 'InvalidMessageSourceAccount':
+          // PROVIDER_ID is not set correctly
+          // Pause publishing entirely
```

Comment:
Do we need to fail the job also? So we know what is happening? if we have multiple workers and only some of them are configured wrong then this would pause the entire queue which is not great